core/basejob.py

```diff
@@ -204,6 +239,7 @@ def postrun(self) -> None:
 
     # =======================================================================
 
+    @_fail_on_exception
```

Comment:
Are we sure we want to have the try block so far on the outside? Now it basically includes a lot of "base class work" like registering the Job with the `JobManager`. My feeling is that if something goes wrong in that generic part, it is likely a PLAMS bug and not an issue with the actual job being run. We also don't want to be so liberal with the exception catching that PLAMS bugs are just flagged as failed jobs. IMHO the job should only fail if there was an issue with the job itself.

Reply:
It's a fair point. Could just wrap the the `prerun` and `postrun` methods, but I guess the issue here is how do we want a PLAMS bug to manifest? At the moment with a parallel job runner, an exception will not get flagged anyway to the user and the job will hang indefinitely. 

Now such a bug would flag as a failed job, and the error message would at least contain the exception message. My concern atm is that PLAMS is very fragile to deadlocking when any exception is raised

Reply:
Alright!